fix: add warning when ignore repo flag

shfc opened the pull request:
<!--  Thanks for sending a pull request!  Here are some tips for you:
1. Make sure to read the Contributing Guide before submitting your PR: https://github.com/helm/helm/blob/main/CONTRIBUTING.md
2. If this PR closes another issue, add 'closes #<issue number>' somewhere in the PR summary. GitHub will automatically close that issue when this PR gets merged. Alternatively, adding 'refs #<issue number>' will not close the issue, but help provide the reviewer more context.-->

**What this PR does / why we need it**:

While there is an argument about treating it as a bug or a UX improvement for this issue, we could add a warning for this unexpected behaviour instead of doing nothing. This makes it easier for others to identify the issue in the logs when they encounter installation problems, saving time and effort. It also gives a heads-up to anyone who might be unknowingly relying on this unexpected behaviour, giving them time to adapt before future upgrades.

refs #7862 

**Special notes for your reviewer**:

**If applicable**:
- [ ] this PR contains user facing changes (the `docs needed` label should be applied if so)
- [ ] this PR contains unit tests
- [x] this PR has been tested for backwards compatibility

shfc committed:
fix: add warning when ignore repo flag

Signed-off-by: Feng Cao <24779889+shfc@users.noreply.github.com>

TerryHowe commented on pkg/action/install.go:
```diff
@@ -759,6 +759,11 @@ func (c *ChartPathOptions) LocateChart(name string, settings *cli.EnvSettings) (
+		// Issue #7862: Helm prioritizes local charts over --repo flag.
+		// This behavior is maintained for backwards compatibility but with a warning.
+		if c.RepoURL != "" {
+			fmt.Printf("WARNING: local chart %s found in current working directory. --repo flag will be ignored\n", name)
```
Maybe something along the lines of
```suggestion
			slog.Warn("local chart found in current working directory. --repo flag ignored", "chart", name)
```

shfc replied:
updated and tested

shfc committed:
fix: replace fmt warning with slog

Signed-off-by: Feng Cao <24779889+shfc@users.noreply.github.com>

TerryHowe reviewed (commented):
/lgtm

Maybe @robertsirc has some opinions on the text logged since he's been working on logging

godhanipayal commented on pkg/action/install.go:
```diff
@@ -759,6 +760,11 @@ func (c *ChartPathOptions) LocateChart(name string, settings *cli.EnvSettings) (
+		if c.RepoURL != "" {
+			slog.Warn("local chart found in current working directory. --repo flag ignored", "chart", name)
+		}
 		return abs, nil
```
Repo flag sometimes adds to the confusion. The above warning looks good. Perhaps, we could add a more actionable warning for clarity
slog.Warn("Local chart found in current working directory; --repo flag ignored. To use the remote chart, remove or rename the local directory.", "chart", name)

shfc replied:
Thanks for the thoughtful suggestion! I reviewed the existing warning log practices in this repo and noticed two general styles: some logs simply state what’s happening, while others provide actionable guidance. It’s not entirely clear to me when one style is preferred over the other.

@robertsirc, do you have any thoughts on which approach aligns best with our current logging conventions?

shfc committed:
fix: move warning to top of block

Emit the warning first to ensure it's always logged. This clarifies that any following errors are due to using a local chart instead of a remote repository.

Signed-off-by: Feng Cao <24779889+shfc@users.noreply.github.com>

shfc commented:
I have made a small change to move the warning to the top of the block, so it’s always logged.  This will make it clear if any subsequent errors are because a local chart is being used instead of a remote repository.

shfc committed:
chore: update generalization warning message

Signed-off-by: Feng Cao <24779889+shfc@users.noreply.github.com>

shfc commented on pkg/action/install.go:
```diff
@@ -750,6 +751,12 @@ func (c *ChartPathOptions) LocateChart(name string, settings *cli.EnvSettings) (
+		// Issue #7862: Helm prioritizes local charts over --repo flag.
+		// This behavior is maintained for backwards compatibility but with a warning.
+		if c.RepoURL != "" {
+			slog.Warn("local chart found in current working directory. --repo flag ignored", "chart", name)
```
Thanks for your suggestion. I have updated it to be more general.

TerryHowe reviewed (approved):
/lgtm

robertsirc reviewed (approved):
LGTM

mattfarina reviewed (approved):
lgtm
